lib/services/staking/reliquary.service.ts

```diff
@@ -270,12 +271,11 @@ export class ReliquaryService {
         const newLevelProgress =
             newLevel > maturityLevels.length ? 'max level reached' : `${newMaturity}/${maturityLevels[newLevel + 1]}`;
 
-        const newLevelProgressDiff = parseInt(maturityLevels[newLevel + 1].toString()) - newMaturity;
-        const oldLevelProgressDiff = parseInt(maturityLevels[levelOnUpdate + 1].toString()) - maturity;
-        const progressDiff = newLevelProgressDiff - oldLevelProgressDiff;
-        const levelDiff = levelOnUpdate - newLevel;
-        const diff = progressDiff + levelDiff * 604800;
-        const diffDate = new Date((nowTimestamp + diff) * 1000);
+        const maxLevel = 10; // zero based indexing, frontend will show '11'
+        const secondsToMaxLevel = parseInt(maturityLevels[maxLevel].toString()) - newMaturity;
+        const diffDate = new Date((nowTimestamp + secondsToMaxLevel) * 1000);
+
+        const staysMax = levelOnUpdate === maxLevel && newLevel === maxLevel;
```

Comment:
Im assuming this variable indicates whether a user is max level or not.
In that case can I suggest making it a bit more explicit, i.e. `isMaxLevel` since it is a boolean value.

Reply:
The variable should indicate the following: "I have a max level relic. I want to deposit X amount into it and the calculation's result is that the relic will stay at max level" (so no need to show the message)